### ISSUE TITLE
Updating the "Fully Bayesian Inference..." example notebook

### ISSUE THREAD

**Issue by jowch** [GPflow/GPflow#988]
... _Provide documentation link:_
https://gpflow.readthedocs.io/en/latest/notebooks/mcmc.html

... _Describe the documentation issue._
I was working through the example notebook linked above and found some things to not work as written. I have noted them below. I'm really excited to try out this project for my GP stuff (currently using MATLAB).

1. The way that `Sum`ed kernels are stored is different from how the notebook describes it. It looks like each of the kernels being summed is now stored in a list of kernels.
```python
# In [4]:
# currently in notebook
m.kern.matern32.lengthscales.prior = gpflow.priors.Gamma(1., 1.)
m.kern.matern32.variance.prior = gpflow.priors.Gamma(1.,1.)
m.kern.bias.variance.prior = gpflow.priors.Gamma(1.,1.)

# should probably be...
m.kern.kernels[0].lengthscales.prior = gpflow.priors.Gamma(1, 1)
m.kern.kernels[0].variance.prior = gpflow.priors.Gamma(1, 1)
m.kern.kernels[1].variance.prior = gpflow.priors.Gamma(1, 1)
```
2. When we are sampling from `s = gpflow.train.HMC()`, the notebook describes that there is a verbose option for `s.sample()`, which is currently broken. This is mentioned in the markdown above this code cell but I wasn't able to find an issue for this.
```python
# In [5]:
# currently in notebook
m.compile()
o = gpflow.train.AdamOptimizer(0.01)
o.minimize(m, maxiter=15) # start near MAP

s = gpflow.train.HMC()
samples = s.sample(m, 100, epsilon=0.12, lmax=20, lmin=5, thin=5, logprobs=False)#, verbose=True)

# should probably be... or fixed
samples = s.sample(m, 100, epsilon=0.12, lmax=20, lmin=5, thin=5, logprobs=False)
```
3. Following from (1), the histogram of the samples of the bias term (are these the hyperparameter values?) should be fixed to account for the change in kernel API.
```python
# In [5]:
# currently in notebook
samples['GPMC/kern/bias/variance'].hist(bins=20)

# should probably be...
samples['GPMC/kern/kernels/1/variance'].hist(bins=20)
```

As far as making these changes goes, is the change for (1) and (3) here to stay? Should we make an issue for (2)? Looking forward to seeing how this goes.

**Comment by st--** [comment]
Hi @jowch,
could you please let us know where you found the link to the `.../latest/...` on readthedocs ? That is an old out-of-date version, which is why it's out of date - you might have noticed the "v:latest" in the bottom left being highlighted in red. The maintained documentation versions are "master" and "develop". https://gpflow.readthedocs.io/en/master/notebooks/mcmc.html has all the fixes that you describe. In `develop`, we actually changed the notebooks around a lot, you can find them here: https://nbviewer.jupyter.org/github/GPflow/GPflow/blob/develop/doc/source/notebooks/intro.ipynb?flush_cache=true (I just noticed it still has the commented-out `verbose` flag though).
Yes, the changes of how sub-kernels are addressed are here to stay!

**Comment by st--** [comment]
PS- forgot to say, but of course thank you for wanting to improve our documentation. :-)

**Comment by jowch** [comment]
Thanks for the clarification! It's actually one of the top hits from Google when you search for gpflow.  Not sure how you would fix this though.

<img width="899" alt="Screen Shot 2019-06-20 at 9 08 54 PM" src="https://user-images.githubusercontent.com/19571703/59897097-f28e3580-939f-11e9-8cc2-324d63bcfed7.png">

**Comment by st--** [comment]
Hi @jowch, thanks for letting us know (I never google for gpflow;))!

We've recently released gpflow-1.4, which contains all the tutorials - since then, `develop` received a couple of minor fixes & improvements only, which we also intend to release fairly soon. We may break things in `develop`, so if you want to make sure your code always keeps running exactly as you had it last time, you might be better off with sticking to a released version - but if you are happy to change your code to accommodate backwards-incompatible changes, you'll benefit from using the develop branch directly by getting the latest bugfixes & improvements! (Sometimes it does take us a while to get around to put a release together, we're not doing this as a fulltime job, and all the developers work off `develop` so don't feel the need for making a release as much.....)
